app/exec/build/tasks/sign.ts

```diff
@@ -0,0 +1,159 @@
+import admZip = require("adm-zip");
+import archiver = require("archiver");
+import del = require('del');
+import fs = require("fs");
+var ncp = require('ncp').ncp;
+import os = require('os');
+import path = require("path");
+import shell = require("shelljs");
+import tasksBase = require("./default");
+import trace = require("../../../lib/trace");
+
+export interface TaskSignResult {
+	signingSuccessful: boolean;
+}
+
+export function getCommand(args: string[]): BuildTaskSign {
+	return new BuildTaskSign(args);
+}
+
+export class BuildTaskSign extends tasksBase.BuildTaskBase<TaskSignResult> {
+	protected description = "Sign a task.";
+
+	constructor(args: string[]) {
+		super(args);
+	}
+
+	protected getHelpArgs(): string[] {
+		return ["certFingerprint", "taskPath", "newGuid", "newNameSuffix"];
+	}
+
+	// tfx build tasks sign --task-path {TASK_PATH} --cert-fingerprint {FINGERPRINT} --new-guid {GUID} --new-name-suffix ' - SIGNED'
+	// --new-guid and --new-name-suffix are optional
+	public async exec(): Promise<TaskSignResult> {
+		const taskZipsPath: string[] | null = await this.commandArgs.taskPath.val();
+		const taskZipPath = taskZipsPath[0];
+		const certFingerprint: string | null = await this.commandArgs.certFingerprint.val();
+		const newGuid: string | null = await this.commandArgs.newGuid.val();
+		const newNameSuffix: string | null = await this.commandArgs.newNameSuffix.val();
+
+		// verify that we can find NuGet
+		const nuGetPath: string = shell.which('nuget');
+		if (!nuGetPath) {
+			throw new Error('Unable to find NuGet. Please add NuGet to the PATH before continuing.');
+		}
+
+		const tempFolder: string = fs.mkdtempSync(path.join(os.tmpdir(), 'task-'));
+		let taskTempFolder: string = path.join(tempFolder, 'task');
+		const taskTempZipPath: string = path.join(tempFolder, 'task.zip');
+		const taskTempNupkgPath: string = path.join(tempFolder, 'task.nupkg');
+
+		// Create temp folder
+		if (fs.existsSync(tempFolder)) {
+			await del(tempFolder, { force: true });
+		}
+
+		fs.mkdirSync(tempFolder);
+		fs.mkdirSync(taskTempFolder);
+
+		// Copy task contents to temp folder
+		shell.cp('-R', taskZipPath, taskTempFolder);
+
+		// Task temp folder is now task\PREVIOUS_NAME
+		taskTempFolder = path.join(taskTempFolder, path.basename(taskZipPath));
+
+		// Update guid/name
+		const needToUpdateTaskJson: boolean = !!newGuid || !!newNameSuffix;
+		if (needToUpdateTaskJson) {
+			const taskJsonPath: string = path.join(taskTempFolder, 'task.json');
+			const data: string = fs.readFileSync(taskJsonPath, 'utf8');
+			let taskJson: any = JSON.parse(data);
+
+			if (newGuid) {
+				taskJson.id = newGuid;
+			}
+
+			if (newNameSuffix) {
+				taskJson.name = `${taskJson.name}${newNameSuffix.replace(/'/g, "")}`;
```

Comment:
Should we just escape `'` to `\'` instead of `"`?

Reply:
This is getting rid of `'` and replacing it with empty string.

Otherwise you get:

`Current task name '- SIGNED'`

Instead of

`Current task name - SIGNED`